### PR TITLE
fix(*): remove utility classes from docs [KHCP-8146]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Addresses: https://konghq.atlassian.net/browse/KHCP-8146

Replaces all usage of Kongponents utilities:
- utility classes (e.g. `w-100, mb-4, d-flex`, etc.)
- provides hardcoded value fallbacks for color/type/spacing variables and removes functions

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [ ] **Tests pass:** check the output of yarn test
* [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [ ] **Framework style:** abides by the essential rules in Vue's style guide
* [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

**Comment by adamdehaven** [comment]
@portikM can you wait to merge this until I can take a close look tomorrow/Friday? 

I want to enable some additional rules along with this merge

**Comment by adamdehaven** [comment]
Derp. Just saw this is for the docs. Ignore my previous comment

**Comment by portikM** [comment]
@adamdehaven any idea why Netlify preview is failing?

File: src/styles/forms/_general.scss [hunk 1]
```diff
@@ -1,19 +1,19 @@
 .form-group {
-  $borderColor: var(--KInputBorder, var(--grey-300, color(grey--grey-300)));
+  $borderColor: var(--KInputBorder, var(--grey-300, #e7e7ec));
   display: block;
-  margin-bottom: var(--lg, spacing(lg));
+  margin-bottom: var(--lg, 24px);
```
Comment (adamdehaven) [inline src/styles/forms/_general.scss:4]: do we need this `--lg` property? Doesn't seem standard and I don't see that we document it anywhere. Same for `--xl` etc.

File: src/styles/forms/_inputs.scss [hunk 2]
```diff
@@ -5,25 +5,25 @@
...
+    color: var(--KInputFocus, var(--blue-500, #1155cb));
     transition: color 0.1s ease;
   }
 
   .is-required {
-    color: var(--KLabelRequiredAsteriskColor, var(--KInputBorder, var(--grey-600)));
+    color: var(--KLabelRequiredAsteriskColor, var(--KInputBorder, var(--grey-600, #3c4557)));
     font-size: 11px;
     font-weight: 500;
     margin-left: var(--spacing-xxs);
```
Comment (adamdehaven) [inline src/styles/forms/_inputs.scss:21]: Can you ensure all of the old custom properties have a fallback value?

File: src/styles/forms/_inputs.scss [hunk 3]
```diff
@@ -84,23 +84,23 @@
     border: none;
     border-radius: 3px;
     box-sizing: border-box;
-    color: var(--KInputColor, var(--black-70, color(black-70)));
+    color: var(--KInputColor, var(--black-70, rgba(0, 0, 0, 0.7)));
     display: block;
-    font-size: var(--type-md, type(md));
+    font-size: var(--type-md, 16px);
     font-weight: 400;
```
Comment (adamdehaven) [inline src/styles/forms/_inputs.scss:90]: question: are you not replacing all of these type of values as well?

File: src/styles/forms/_inputs.scss (hunk 2 above)
Comment (portikM) [reply src/styles/forms/_inputs.scss:21]: Went through them all to make sure they have fallback values

File: src/styles/forms/_inputs.scss (hunk 3 above)
Comment (adamdehaven) [reply src/styles/forms/_inputs.scss:90]: Right but why not use a token..?

File: docs/.vitepress/theme/_variables.scss [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
+  --vp-c-brand: #1155cb;
+  --vp-c-brand-light: #3972d5;
+  --vp-c-brand-lighter: #8ab3fa;
+  --vp-c-brand-dark: #003694;
+  --vp-c-brand-darker: #0a2b66;
   // Homepage site name color
   --kongponents-theme-background-gradient: linear-gradient(120deg, #169fcc, #42d782);
   --vp-home-hero-name-color: transparent;
-  --vp-home-hero-name-background: var(--kongponents-theme-background-gradient, var(--vp-c-brand, #1456cb));
+  --vp-home-hero-name-background: var(--kongponents-theme-background-gradient, #1456cb);
```
[the remark this answers is not in the document]
Comment (portikM) [reply docs/.vitepress/theme/_variables.scss]: Fixed

File: src/styles/forms/_inputs.scss (hunk 3 above)
Comment (portikM) [reply src/styles/forms/_inputs.scss:90]: Ah sorry, I misunderstood your question. Styles in this specific file will be moved to the component/mixins soon - that's when I was planning to go though all CSS rules line-by-line (you can notice that this file has a lot more non-tokenized rules). That's why didn't bother inserting the token here initially. I fixed it now but going further ignore hardcoded/non-token values in these standalone files.

Comment (adamdehaven) [reply src/styles/forms/_inputs.scss:90]: Ah ok makes sense; thanks for the context

File: docs/.vitepress/theme/_overrides.scss [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
 .vp-doc {
-  --vp-custom-block-info-border: var(--blue-600, #003694);
-  --vp-custom-block-info-text: var(--blue-500, #1155cb);
-  --vp-custom-block-info-bg: var(--blue-100, #f2f6fe);
+  --vp-custom-block-info-border: #003694;
+  --vp-custom-block-info-text: #1155cb;
+  --vp-custom-block-info-bg: #f2f6fe;
 
   .kong-card li + li {
     margin-top: 0;
```
Comment (DariaYeremina) [inline docs/.vitepress/theme/_overrides.scss:8]: nit: $kui-space-0

File: docs/components/card.md [hunk 6]
```diff
@@ -179,26 +175,33 @@ Cards can be arranged with flex box.
 </div>
 
 ```html
-<div class="d-flex flex-row">
-  <KCard title="Left" class="w-auto" body="This card only has a title" />
-  <KCard title="Center" class="w-auto mx-5" body="This card always has a icon button">
+<div class="">
```
Comment (DariaYeremina) [inline docs/components/card.md:178]: suggestion: remove empty `class` attr

File: docs/components/card.md [hunk 7]
```diff
@@ -145,31 +145,27 @@ Set if the card should always have shadow state (shadow)
...
     <template v-slot:actions>
       <KButton size="small" appearance="outline">
         <KIcon
-            icon="gearFilled"
-            size="16"
-            view-box="0 0 16 16"
-            class="pr-0"
-          />
+          icon="gearFilled"
+          size="16"
```
Comment (DariaYeremina) [inline docs/components/card.md:161]: suggestion: use icon size var

File: docs/components/card.md [hunk 8]
```diff
@@ -179,26 +175,33 @@ Cards can be arranged with flex box.
...
     <template v-slot:actions>
       <KButton>
         <KIcon
-            icon="gearFilled"
-            size="16"
-            view-box="0 0 16 16"
-            class="pr-0"
-          />
+          icon="gearFilled"
+          size="16"
```
Comment (DariaYeremina) [inline docs/components/card.md:185]: suggestion: use icon size var

File: docs/components/card.md [hunk 9]
```diff
@@ -214,9 +217,10 @@ Cards can be arranged with flex box.
   <template v-slot:statusHat>
     <KIcon
       icon="check"
+      class="horizontal-spacing"
       color="#07A88D"
-      class="mr-2"
-      size="24" />
+      size="24"
```
Comment (DariaYeremina) [inline docs/components/card.md:222]: suggestion: use icon size var

File: docs/components/card.md [hunk 10]
```diff
@@ -236,7 +240,6 @@ Cards can be arranged with flex box.
     <KIcon
       icon="check"
       color="#07A88D"
-      class="mr-2"
       size="24" />
```
Comment (DariaYeremina) [inline docs/components/card.md:243]: suggestion: use icon size var

File: docs/components/prompt.md [hunk 11]
```diff
@@ -209,28 +209,28 @@ There are 3 designated slots you can use to display content in the modal.
...
-    <KButton appearance="outline" @click="slotsIsOpen = false" class="non-visual-button">Close</KButton>
+    <KButton appearance="outline" @click="slotsIsOpen = false">Close</KButton>
   </template>
 </KPrompt>
 
 ```html
 <KPrompt :is-visible="slotsIsOpen" @canceled="slotsIsOpen = false" @proceed="slotsIsOpen = false">
   <template v-slot:header-content>
-    <KIcon icon="immunity" color="#7F01FE" class="mr-2" size="20" />
+    <KIcon icon="immunity" color="#7F01FE" size="20" />
```
Comment (DariaYeremina) [inline docs/components/prompt.md:226]: suggestion: use icon size var

File: docs/components/segmented-control.md [hunk 12]
```diff
@@ -236,24 +236,24 @@ An Example of changing the KSegmentedControl to a purple theme instead of blue m
...
 <style>
 .purple-segment {
-  --KSegmentedControlText: var(--purple-400, #{$kui-method-color-text-connect});
-  --KSegmentedControlSelectedBackground: var(--white, #{$kui-color-background});
-  --KSegmentedControlSelectedBorder: var(--purple-300, #9396fc);
-  --KSegmentedControlUnselectedBackground: var(--blue-100, #{$kui-color-background-primary-weakest});
-  --KSegmentedControlUnselectedBorder: var(--purple-200, #bec0fd);
-  --KSegmentedControlGap: var(--spacing-sm, #{$kui-space-50});
+  --KSegmentedControlText: #473cfb;
+  --KSegmentedControlSelectedBackground: white;
```
Comment (DariaYeremina) [inline docs/components/segmented-control.md:240]: suggestion: use background color var

File: docs/index.md [hunk 13]
```diff
@@ -42,7 +42,7 @@ features:
 /* Homepage Components button */
 .VPContent.is-home .VPButton.medium.alt[href^="/components/"] {
   color: #fff;
```
Comment (DariaYeremina) [inline docs/index.md:44]: suggestion: use text color var

File: src/styles/_mixins.scss [hunk 14]
```diff
@@ -22,58 +22,58 @@
 }
 
 @mixin input-default {
-  background-color: var(--KInputBackground, var(--white, color(white)));
-  box-shadow: inset 0 0 0 1px var(--KInputBorder, var(--grey-300)) !important;
+  background-color: var(--KInputBackground, var(--white, #ffffff));
```
Comment (DariaYeremina) [inline src/styles/_mixins.scss:25]: question: can we use `$kui-color-background` here?

File: src/styles/_mixins.scss [hunk 15]
```diff
@@ -22,58 +22,58 @@
...
 @mixin input-focus {
-  box-shadow: inset 0 0 0 1px var(--KInputFocus, var(--blue-400)) !important;
+  box-shadow: inset 0 0 0 1px var(--KInputFocus, var(--blue-400, #3972d5)) !important;
   outline: none !important;
   transition: all 0.1s ease;
 }
 
 @mixin textarea-default {
-  background-color: var(--KInputBackground, var(--white, color(white)));
+  background-color: var(--KInputBackground, var(--white, #ffffff));
```
Comment (DariaYeremina) [inline src/styles/_mixins.scss:37]: question: can we use $kui-color-background here?

File: src/styles/_mixins.scss [hunk 16]
```diff
@@ -22,58 +22,58 @@
...
+  box-shadow: inset 0 0 0 1px var(--KInputBorder, var(--grey-300, #e7e7ec)) !important;
   cursor: not-allowed;
   font-style: italic;
   transition: all 0.1s ease;
 }
 
 @mixin fullscreen-loading-container {
   align-items: center;
-  background: var(--white, color(white));
+  background: var(--white, #ffffff);
```
Comment (DariaYeremina) [inline src/styles/_mixins.scss:76]: question: can we use $kui-color-background here?

File: src/styles/forms/_general.scss [hunk 17]
```diff
@@ -1,19 +1,20 @@
 .form-group {
-  $borderColor: var(--KInputBorder, var(--grey-300, color(grey--grey-300)));
+  $borderColor: var(--KInputBorder, var(--grey-300, #e7e7ec));
   display: block;
-  margin-bottom: var(--lg, spacing(lg));
+  margin-bottom: 24px;
```
Comment (DariaYeremina) [inline src/styles/forms/_general.scss:4]: nit: $kui-space-80

File: src/styles/forms/_general.scss [hunk 18]
```diff
@@ -1,19 +1,20 @@
...
   width: 100%;
 
   hr {
     border-color: $borderColor;
-    margin: var(--xl, spacing(xl)) 0;
+    margin: 32px 0;
```
Comment (DariaYeremina) [inline src/styles/forms/_general.scss:9]: nit: $kui-space-90 $kui-space-0

File: src/styles/forms/_inputs.scss [hunk 19]
```diff
@@ -139,7 +139,7 @@
 
     /* Browser Overrides */
     &::placeholder {
-      color: var(--KInputPlaceholderColor, var(--black-45, color(black-45)));
+      color: var(--KInputPlaceholderColor, var(--black-45, rgba(0, 0, 0, 0.45)));
       font-weight: 400;
```
Comment (DariaYeremina) [inline src/styles/forms/_inputs.scss:143]: nit: $kui-font-weight-regular

File: src/styles/forms/_switch.scss [hunk 20]
```diff
@@ -38,7 +38,7 @@ $transition: 0.2s linear;
 
     // Toggle
     &:after {
-      background-color: var(--white, color(white));
+      background-color: var(--white, #ffffff);
```
Comment (DariaYeremina) [inline src/styles/forms/_switch.scss:41]: nit: $kui-color-background

File: src/styles/_mixins.scss (hunk 14 above)
Comment (portikM) [reply src/styles/_mixins.scss:25]: Same as with one of Adam's questions: tokens in standalone styles files (`mixins.scss, inputs.scss, etc.`) will be replaced in a separate ticket

File: src/styles/_mixins.scss (hunk 15 above)
Comment (portikM) [reply src/styles/_mixins.scss:37]: See above

File: docs/components/button.md [hunk 21]
```diff
@@ -186,17 +190,19 @@ Should you need to use a KTooltip component on a KButton with `disabled` attribu
...
-</KButton>
-<KButton appearance="secondary" size="small">
-  <template v-slot:icon>
-    <KIcon icon="gear" color="var(--KButtonSecondaryColor, #003694)"/>
-  </template>
-</KButton>
+<div class="spacing-container">
+  <KButton appearance="secondary">
+    <template v-slot:icon>
+      <KIcon icon="externalLink" color="var(--KButtonSecondaryColor, #003694)"/>
```
[the remark this answers is not in the document]
Comment (portikM) [reply docs/components/button.md:196]: Eh probably not, removed it

File: docs/components/card.md (hunk 7 above)
Comment (portikM) [reply docs/components/card.md:161]: This is docs file, it's fine to use hardcoded values here (not in the components though). Same for other places

File: docs/components/popover.md [hunk 22]
```diff
@@ -678,9 +678,9 @@ export default defineComponent({
 <KPop @opened="loadSomething" @closed="onClose">
   <KButton :disabled="currentState == 'pending'">{{ buttonText }}</KButton>
   <template v-slot:content>
-    <div style="justify-content: center;" class="d-flex">
+    <div style="justify-content: center;">
       <KIcon v-if="currentState == 'pending'" icon="spinner" color="purple" />
-      <div class="pl-2" style="line-height: 24px;">{{ message }}</div>
+      <div style="line-height: 24px;">{{ message }}</div>
```
[the remark this answers is not in the document]
Comment (portikM) [reply docs/components/popover.md]: This can be removed

File: docs/components/segmented-control.md (hunk 12 above)
Comment (portikM) [reply docs/components/segmented-control.md:240]: This is docs file, it's fine to hardcode some colors here.

File: docs/components/segmented-control.md [hunk 23]
```diff
@@ -236,24 +236,24 @@ An Example of changing the KSegmentedControl to a purple theme instead of blue m
...
 <style scoped lang="scss">
 .purple-segment {
-  --KSegmentedControlText: var(--purple-400, #{$kui-method-color-text-connect});
-  --KSegmentedControlSelectedBackground: var(--white, #{$kui-color-background});
-  --KSegmentedControlSelectedBorder: var(--purple-300, #9396fc);
-  --KSegmentedControlUnselectedBackground: var(--blue-100, #{$kui-color-background-primary-weakest});
-  --KSegmentedControlUnselectedBorder: var(--purple-200, #bec0fd);
-  --KSegmentedControlGap: var(--spacing-sm, #{$kui-space-50});
+  --KSegmentedControlText: #473cfb;
+  --KSegmentedControlSelectedBackground: white;
```
[the remark this answers is not in the document]
Comment (portikM) [reply docs/components/segmented-control.md:252]: Same here.

File: docs/index.md (hunk 13 above)
Comment (portikM) [reply docs/index.md:44]: I tried and for some reason `$kui-` SCSS variables don't work in this file. Either way we can tackle this later.

File: docs/components/card.md (hunk 7 above)
Comment (DariaYeremina) [reply docs/components/card.md:161]: I saw using var in other .md file in this pr

Comment (portikM) [reply docs/components/card.md:161]: Maybe but this PR mostly replaces:
- utility classes (e.g. w-100, mb-4, d-flex, etc.)
- provides hardcoded value fallbacks for color/type/spacing variables and removes functions

I can handle changing icon size in separate PR, it's not in the scope for this one.

File: docs/components/card.md (hunk 6 above)
Comment (portikM) [reply docs/components/card.md:178]: Done

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**